src/ProfessorPopup.tsx

```diff
@@ -19,34 +25,269 @@ interface professorPopupTooltipProps {
+ * @returns See ProfessorInfo interface for object returned
+ */
+async function professorRequest(
+  professor: string
+): Promise<ProfessorInfo | string> {
+  const url = 'https://api.cppbroncodirect.me/professor';
+  const request = await fetch(url, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ name: ProfessorNameFiltering(professor) }),
+  });
+  if (!(await request.ok)) {
+    return await request.text();
+  }
+  return await request.json();
+}
+
+/**
+ * Component that shows the info inside the popup
+ * @param props See professorPopupTooltipProps interface (control click) to see parameters
+ * @returns Element containing the UI for professor information
+ */
+function ProfessorPopupInfo(props: professorPopupTooltipProps): JSX.Element {
+  const [professorData, setProfessorData] = useState({
+    difficulty: 'N/A', // avgDifficulty
+    rating: 'N/A', // avgRating
+    reviews: 'N/A', // numRatings
+    retention: 'N/A', // wouldTakeAgainPercent
+  });
+
+  const [loading, setLoading] = useState(false);
+  const [hasResult, setHasResult] = useState(true);
+  const [page, setPage] = useState(0);
+  const [currentProfessor, setCurrent] = useState('');
+  const filteredProfNames = ProfessorNameFiltering(props.professorName);
+  const professorsList = filteredProfNames.split('&');
+  const [errorMessage, setErrorMessage] = useState(
+    'Something went wrong when trying to retrieve data'
+  );
+
+  /**
+   * Gets professor data from backend 'server.ts/professor' function and sets data to their respective useState
+   */
+  async function getProfessorData(): Promise<void> {
+    try {
+      setHasResult(true);
+      const selectedProf = professorsList[page].trim();
+      setCurrent(selectedProf);
+      const request = await professorRequest(selectedProf);
+      // Throws an error if request doesn't return valid RMP info
+      if (request === 'professor not found in mapping') {
+        throw Error('Professor not found on RMP');
+      }
+
+      // @ts-expect-error
+      // Suppresses TypeScript error saying that this operation cannot be done if request is a string
+      const {
+        avgDifficulty,
+        avgRating,
+        numRatings,
+        wouldTakeAgainPercent,
+      }: ProfessorInfo = request;
+
+      setProfessorData({
+        difficulty: numRatings > 0 ? avgDifficulty : 'N/A', // if there are 0 reviews, there can't be any data
+        rating: numRatings > 0 ? avgRating : 'N/A',
+        reviews: numRatings > 0 ? numRatings.toString() : 'N/A',
+        retention:
+          numRatings > 0 || wouldTakeAgainPercent < 0
+            ? wouldTakeAgainPercent.toString()
+            : 'N/A', // RMP can return wouldTakeAgainPercent as -1
+      });
+
+      setLoading(true); // data finished loading
+    } catch (error) {
+      if (error instanceof Error) {
+        setErrorMessage('This professor does not have a RateMyProfessor page.');
```

Comment:
Indicate which professor caused this error for clarity (i.e., TBA/[insert professor here])
- e.g., The professor **Some Name** does not have a RateMyProfessor page.

If possible, try to put the entire text on 2 lines so the popup doesn't span a big length.
![image](https://user-images.githubusercontent.com/24869207/228996521-119c7703-728e-4b63-a0f2-a53a150b60dc.png)